### PR TITLE
feat(text-field): Full-width support

### LINKED ISSUE #4479: Implement Text Field v2 Component
Spec reference: https://material.google.com/components/text-fields.html

- [x] Single-line
- [x] Multi-line (`<textarea>`?)
- [x] Full-width
- ~~Character Counter~~ (split into #4823)

This is our `<input type="text">`.

Auto-complete is not included in the scope of this feature.

### PR DESCRIPTION
**Pull request by traviskaufman**
* Add styles for full-width text fields
* Make JS Textfield component more resiliant to label omission
* Fix issue where multi-line textfields that were upgraded had double
  bottom borders
* Use better data attr name for excluding auto-initialization of text
  fields within the demo

Closes #4479
[Delivers #129705931]

### CODE REVIEW

**Comment by Garbee** [comment]
Why is a full width textfield not simply the default while developers need to manually set the width if they want it fixed for some reason?

**Comment by traviskaufman** [comment]
@Garbee The spec seems to consider the interactive, single-line input fields as the "default", whereas the full width style text fields are used for more specialized cases. So I'm try

File: demos/textfield.html [hunk 1]
```diff
@@ -236,5 +259,33 @@
...
+          var target = evt.target;
+          [tf, tfMulti].forEach(function(component) {
+            component.disabled = target.checked;
+          });
+        });
+
+        document.getElementById('fullwidth-dense').addEventListener('change', function(evt) {
+          var target = evt.target;
+          [tfRoot, tfMultiRoot].forEach(function(el) {
+            el.classList[target.checked ? 'add' : 'remove']('mdl-textfield--dense');
```
Comment (sgomes) [inline demos/textfield.html]: Wow, neat shorthand!

File: packages/mdl-textfield/index.js [hunk 2]
```diff
@@ -58,8 +58,18 @@ export default class MDLTextfield extends MDLComponent {
     return new MDLTextfieldFoundation(Object.assign({
       addClass: className => this.root_.classList.add(className),
       removeClass: className => this.root_.classList.remove(className),
-      addClassToLabel: className => this.label_.classList.add(className),
-      removeClassFromLabel: className => this.label_.classList.remove(className)
+      addClassToLabel: className => {
+        const {label_: label} = this;
```
Comment (sgomes) [inline packages/mdl-textfield/index.js]: Why not just
```js
if (this.label_) {
  this.label_.classList.add(className);
}
```
?

File: packages/mdl-textfield/index.js [hunk 3]
```diff
@@ -58,8 +58,18 @@ export default class MDLTextfield extends MDLComponent {
...
+        if (label) {
+          label.classList.add(className);
+        }
+      },
+      removeClassFromLabel: className => {
+        const {label_: label} = this;
```
Comment (sgomes) [inline packages/mdl-textfield/index.js]: Ditto.

File: packages/mdl-textfield/index.js (hunk 2 above)
Comment (traviskaufman) [inline packages/mdl-textfield/index.js]: Changing to `const label = this.label_;` and merging.

**Pull request merged**